From 8ef8f88a00e5a755af63873149db92e193c2b3a0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=B6ren=20Tempel?= <soeren+git@soeren-tempel.net>
Date: Sun, 10 Dec 2023 19:20:03 +0100
Subject: [PATCH] Flush stdout after writing command string to it

Otherwise, if stdout is fully buffered (e.g. because the output is
redirected to a file or pipe) then the command string may still
be buffered when the command is executed in which case the command
string will be written to stdout **after** the command has written
its output to stdout.
---
 make.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/make.c b/make.c
index 44bb95b..bd96309 100644
--- a/make.c
+++ b/make.c
@@ -56,8 +56,10 @@ docmds(struct name *np, struct cmd *cp)
 		} else if (!sdomake)
 			ssilent = dotouch;
 
-		if (!ssilent)
+		if (!ssilent) {
 			puts(q);
+			fflush(stdout);
+		}
 
 		if (sdomake) {
 			// Get the shell to execute it
